osidb/models.py

```diff
@@ -411,21 +411,25 @@ class FlawManager(ACLMixinManager):
     """flaw manager"""
 
     @staticmethod
-    def create_flaw(bz_id=None, cve_id=None, **extra_fields):
+    def create_flaw(bz_id, full_match=False, **extra_fields):
         """
         return a new flaw or update an existing flaw without saving
         this is meant for cases when UUID is not available - collector
-        """
-        # only one case when they're both equal: (None, None)
-        assert cve_id != bz_id, "Either cve_id or bz_id must be provided"
 
+        full match means that we match on both Bugzilla and CVE ID
+        """
         try:
-            flaw = (
-                # CVE-less flaws are identified by Bugzilla ID
-                Flaw.objects.get(meta_attr__bz_id=bz_id)
-                if cve_id is None
-                else Flaw.objects.get(cve_id=cve_id)
-            )
+            cve_id = extra_fields.get("cve_id")
```

Comment:
`cve_id` can be `None`, would this be handled correctly in both scenarios?

Reply:
might require some extra tests

Reply:
This is an expected and covered case. All the test cases where the alias contains no CVEs (either an empty array or some non-CVE alias) deal with it and hope I covered all possible cases here.